lib/identity/realm.js

```diff
@@ -322,7 +358,35 @@ class Realm extends CryptoConsumer {
         fields: resp.fields,
         context: resp.context,
       }
-      const data = await actionHandler(err, actionData)
+      let data
+      switch (resp.type) {
+        case 'continue':
+          data = {}
+          break
+        case 'password-challenge':
+          data = { password }
+          break
+        case 'register-brokered-user':
+          cryptoKeys = await this.storage.generateKeypair()
+          signingKeys = await this.storage.generateSigningKeypair()
+          data = {
+            public_key: cryptoKeys.publicKey,
+            public_signing_key: signingKeys.publicKey,
+          }
+          break
+        case 'complete-broker-registration':
+          await this.registerFederatedUser(
+            username,
+            resp,
+            cryptoKeys,
```

Comment:
aren't cryptokeys and signing keys only valid if the above case is run?
```
        case 'register-brokered-user':
          cryptoKeys = await this.storage.generateKeypair()
          signingKeys = await this.storage.generateSigningKeypair()
```

Reply:
Ah I think I'm understanding the federated login flow, for the future I think pulling it out into a separate method would help with clarity. (doesn't have to be done in this PR)

Reply:
yeah, it's a bit odd -- the way the required action is written these will always be in series. This could be an intermediate object, or something, but we're trying to get this out the door, so I just went with higher scoped vars to pass the data from one action to the next. I'll add some code comments to that effect.

Reply:
I've added some comments to hopefully clarify what is happening a bit. This is definitely a sprawling method that could use some sub-functions to clean up the logic.